### DISCUSSION TITLE
Change Page Title from individual ftl files

### DISCUSSION THREAD

**Discussion by Roballen** [keycloakify/keycloakify#428] [General]
Is it possible to change the page title from individual ftl files? I found a similar discussion, but not quite the same. I looked around both the starter and keycloakify source code and tried a few things without success.

I know I can handle this with js, but that seems suboptimal. For example, let's say my project/theme is MyApp. I want unique page titles (or the designers do) for

MyApp - Login
MyApp - Forgot Password
MyApp - Error
MyApp - Register

When I look template.ftl in the base theme I can see this is handled by theme.preoperties, but since we are using the index.html it doesn't seem like it is possible at this time? Is that correct?

`<title>${msg("loginTitle",(realm.displayName!''))}</title>`

**Comment by garronej** [comment]
Hi @Roballen,  

I'd do something like:  

`src/login/KcApp.ts`
```ts
export function KcApp(props: { kcContext: KcContext; }){

    useEffect(
        ()=> {
        
            document.title = kcContext.pageId;
        
        },
        []
     );

}
```

Hope it helps

**Reply by Roballen** [reply]
sounds good, that was my approach but thought there could be another way. Thanks for the help.
